src/fire/io/h5/Reader.cxx

```diff
@@ -60,6 +66,138 @@ std::vector<std::array<std::string,3>> Reader::availableObjects() {
+   *    read from other DataSets.
+   * 3. Set-up these atomic data sets in a structure that mimics the on-disk structure in-memory.
+   *
+   * ## Every Copy
+   * Connect the Reader::load handles immediately to the Writer::save call.
+   */
+
+  // this is where recursing into the subgroups of full_name occurs
+  // if this mirror object hasn't been created yet
+  if (mirror_objects_.find(path) == mirror_objects_.end()) {
+    mirror_objects_.emplace(std::make_pair(path,
+          std::make_unique<MirrorObject>(path, *this)));
+  }
+  // do the copying
+  mirror_objects_[path]->copy(i_entry, 1, output);
+}
+
+class create_data {
+  const std::string& path_;
+  HighFive::DataType type_;
+  std::unique_ptr<BaseData> data_;
+
+ public:
+  create_data(const std::string& p, HighFive::DataType t)
+    : type_{t}, path_{p} {}
+
+  std::unique_ptr<BaseData> get() {
+    return std::move(data_);
+  }
+
+  template<typename T>
+  void operator()(T) {
+    if (type_ == HighFive::create_datatype<T>()) {
+      if (data_) { char a; std::cerr << "BAD"; std::cin >> a; }
+      else data_ = std::make_unique<io::Data<T>>(path_);
+    } 
+  }
+};
+
+Reader::MirrorObject::MirrorObject(const std::string& path, Reader& reader) 
+  : reader_{reader} {
+  if (reader_.getH5ObjectType(path) == HighFive::ObjectType::Dataset) {
+    // simple atomic event object
+    //  unfortunately, I can't think of a better solution than manually
+    //  copying the code for all of the types
+    HighFive::DataType type = reader_.getDataSetType(path);
+    if (type == HighFive::create_datatype<int>()) {
+      data_ = std::make_unique<io::Data<int>>(path);
+    } else if (type == HighFive::create_datatype<long int>()) {
+      data_ = std::make_unique<io::Data<long int>>(path);
+    } else if (type == HighFive::create_datatype<long long int>()) {
+      data_ = std::make_unique<io::Data<long long int>>(path);
+    } else if (type == HighFive::create_datatype<unsigned int>()) {
+      data_ = std::make_unique<io::Data<unsigned int>>(path);
+    } else if (type == HighFive::create_datatype<unsigned long int>()) {
+      data_ = std::make_unique<io::Data<unsigned long int>>(path);
+    } else if (type == HighFive::create_datatype<unsigned long long int>()) {
+      data_ = std::make_unique<io::Data<unsigned long long int>>(path);
+    } else if (type == HighFive::create_datatype<float>()) {
+      data_ = std::make_unique<io::Data<float>>(path);
+    } else if (type == HighFive::create_datatype<double>()) {
+      data_ = std::make_unique<io::Data<double>>(path);
+    } else if (type == HighFive::create_datatype<std::string>()) {
+      data_ = std::make_unique<io::Data<std::string>>(path);
+    } else if (type == HighFive::create_datatype<fire::io::Bool>()) {
+      data_ = std::make_unique<io::Data<bool>>(path);
+    } else {
+      std::cerr << "BAD: Unable to deduce C++ type from H5 type." << std::endl;
```

Comment:
Need to decide if this is a warning or an exception. I feel like it could be an exception...

```suggestion
      throw Exception("UnknownDS","Unable to deduce C++ type from H5 type during a copy\n"
        "    User could avoid this issue simply by accessing the event object within some processor during the first event.", 
        false);
```